hive/src/main/java/org/apache/iceberg/IcebergSchemaUtil.java

```diff
@@ -30,7 +30,7 @@ public class IcebergSchemaUtil {
   public static PartitionSpec copyPartitionSpec(PartitionSpec partitionSpec, Schema copySchema) {
     PartitionSpec.Builder builder = PartitionSpec.builderFor(copySchema);
     partitionSpec.fields().forEach(partitionField -> {
-      builder.add(partitionField.sourceId(), partitionField.name(), partitionField.transform());
+      builder.add(partitionField.sourceId(), partitionField.name().toLowerCase(), partitionField.transform());
```

Comment:
I suggest clarifying the behavior and reasons for changing partition fields to lowercase in the Java doc.

Reply:
Okay, I'll add some notes